bots/gcal-bot

```diff
@@ -1,86 +1,124 @@
+    access to certain aspects of your Google Account.
 
-    Run this on your personal machine.  Your API key and calendar URL are revealed to local
-    users through the command line.
+    This integration should be run on your local machine. Your API key and other information are
+    revealed to local users through the command line.
 
-    Depends on: python-gdata
+    Depends on: google-api-python-client
 """)
 
-parser.add_option('--calendar',
-                  dest='calendar',
-                  action='store',
-                  help='Google Calendar XML "Private Address"',
-                  metavar='URL')
+
 parser.add_option('--interval',
                   dest='interval',
-                  default=10,
+                  default=30,
                   type=int,
                   action='store',
-                  help='Minutes before event for reminder [default: 10]',
+                  help='Minutes before event for reminder [default: 30]',
                   metavar='MINUTES')
+
+parser.add_option('--calendar',
+                  dest = 'calendarID',
+                  default = 'primary',
+                  type = str,
+                  action = 'store',
+                  help = 'Calendar ID for the calendar you want to receive reminders from.')
+
 parser.add_option_group(zulip.generate_option_group(parser))
 
 (options, args) = parser.parse_args()
 
-if not (options.zulip_email and options.calendar):
-    parser.error('You must specify --user and --calendar')
+if not (options.zulip_email):
+    parser.error('You must specify --user')
 
 try:
-    from gdata.calendar.client import CalendarClient
+    from googleapiclient import discovery
 except ImportError:
-    parser.error('Install python-gdata')
+    parser.error('Install google-api-python-client')
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+CLIENT_SECRET_FILE = 'client_secret.json'
+APPLICATION_NAME = 'Zulip'
 
-def get_calendar_url():
-    # type: () -> str
-    parts = urllib.parse.urlparse(options.calendar)
-    pat = os.path.split(parts.path)
-    if pat[1] != 'basic':
-        parser.error('The --calendar URL should be the XML "Private Address" ' +
-                     'from your calendar settings')
-    return urllib.parse.urlunparse((parts.scheme, parts.netloc, pat[0] + '/full',
-                                    '', 'futureevents=true&orderby=startdate', ''))
+def get_credentials():
+    # type: () -> client.Credentials
+    """Gets valid user credentials from storage.
 
-calendar_url = get_calendar_url()
+    If nothing has been stored, or if the stored credentials are invalid,
+    an exception is thrown and the user is informed to run the script in this directory to get
+    credentials.
 
-client = zulip.init_from_options(options)
+    Returns:
+        Credentials, the obtained credential.
+    """
+    try:
+        credential_dir = os.path.abspath(os.path.join(__file__, '../'))
```

Comment:
May be you need to create a constant called `ZULIP_PATH` and use it to create paths. This applies to path definitions above this as well.